src/renderers/webgl/WebGLCubeMaps.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @author mrdoob / http://mrdoob.com/
+ */
+
+import { CubeReflectionMapping, CubeRefractionMapping, EquirectangularReflectionMapping, EquirectangularRefractionMapping } from '../../constants.js';
+import { WebGLCubeRenderTarget } from "../WebGLCubeRenderTarget.js";
+
+function WebGLCubeMaps( renderer ) {
+
+	let cubemaps = new WeakMap();
+
+	function mapTextureMapping( texture, mapping ) {
+
+		if ( mapping === EquirectangularReflectionMapping ) {
+
+			texture.mapping = CubeReflectionMapping;
+
+		} else if ( mapping === EquirectangularRefractionMapping ) {
+
+			texture.mapping = CubeRefractionMapping;
+
+		}
+
+		return texture;
+
+	}
+
+	function get( texture ) {
+
+		if ( texture && texture.isTexture ) {
+
+			const mapping = texture.mapping;
+
+			if ( mapping === EquirectangularReflectionMapping || mapping === EquirectangularRefractionMapping ) {
+
+				if ( cubemaps.has( texture ) ) {
+
+					const cubemap = cubemaps.get( texture ).texture;
+					return mapTextureMapping( cubemap, texture.mapping );
+
+				} else {
+
+					const image = texture.image;
+
+					if ( image.height > 0 ) {
```

Comment:
@mrdoob @Mugen87

I think this is a problem... `image` can be undefined if the texture has not loaded yet.

What should this function return if the texture is not ready?

Reply:
When looking at the code, there should be no breakage if `get()` returns `undefined` since subsequent code checks for `undefined` or `null`.

That said, it is still better if we add an `else` branch that explicitly returns `null`.